lab1/lab1.rst

```diff
@@ -0,0 +1,412 @@
+   number of inputs, and the data bit width is equal to the number of
+   output bits.
+
+#. Using the input bits as the address, fill in the entries of the ROM
+   with the correct outputs for that combination of inputs.
+
+#. To get the output, address the ROM using the concatenation of the
+   input signals.
+
+Example
+-------
+
+The following is an example of implementing the XOR function using
+microcode. Here is the truth table for XOR
+:math:`\left( \oplus \right)`.
+
++---+---+--------------------+
+| X | Y | X :math:`\oplus` Y |
++---+---+--------------------+
+| 0 | 0 | 0                  |
++---+---+--------------------+
+| 0 | 1 | 1                  |
++---+---+--------------------+
+| 1 | 0 | 1                  |
++---+---+--------------------+
+| 1 | 1 | 0                  |
++---+---+--------------------+
+
+Applying step 1, we first create a ROM with 2 address bits, because we
+have 2 inputs. Each entry is 1 bit wide, because there is only output.
+Next, we fill in the ROM using X and Y as the address bits.
+
++---+---+----------------------------------------+-------+
+| X | Y | Address                                | Value |
++===+===+========================================+=======+
+| 0 | 0 | :math:`00 \left( \mathtt{0x0} \right)` | 0     |
++---+---+----------------------------------------+-------+
+| 0 | 1 | :math:`01 \left( \mathtt{0x1} \right)` | 1     |
++---+---+----------------------------------------+-------+
+| 1 | 0 | :math:`10 \left( \mathtt{0x2} \right)` | 1     |
++---+---+----------------------------------------+-------+
+| 1 | 1 | :math:`11 \left( \mathtt{0x3} \right)` | 0     |
++---+---+----------------------------------------+-------+
+
+The Logisim implementation is included with the given files for the
+assignment, in the subcircuit **ROM XOR**.
+
+Assignment
+==========
+
+#. Implement a combinational circuit using combinational logic.
+
+#. Implement a combinational circuit using a ROM.
+
+For each circuit, please create a sub-circuit with appropriately named
+inputs and outputs.
+
+Combinatorial Circuit - Combinational Logic
+-------------------------------------------
+
+Implement the circuit that has the following truth table using
+**combinational logic**.
+
+-  When creating the combinational circuit using combinational logic,
+   you may only use splitters and these gates: AND, OR, and NOT.
+
+-  The logic should be the minimal amount to express the truth table.
+   Do not worry too much about this -- this will be true if the circuit
+   is generated using Logisim's built-in analysis tool, mentioned below.
+
+-  **In** are the inputs and **Out** are the outputs.
+
+-  All unspecified input and output combinations are don’t cares, as are
+   the **D**\ s in the table.
+
++-----+-----+-----+-----+-----+-----+-----+-----+------+------+------+
+| In7 | In6 | In5 | In4 | In3 | In2 | In1 | In0 | Out2 | Out1 | Out0 |
++=====+=====+=====+=====+=====+=====+=====+=====+======+======+======+
```

Comment:
Can you add a blank column between the In and Out so it's more clear where the transition is? Also, I'm not sure if I asked this, but can you change some of these numbers so people can't turn in the exact same file from last quarter?